src/manifest/deployment/index.ts

```diff
@@ -0,0 +1,174 @@
+import { ResourceCollection } from '../types';
+import { ListChoiceOptions } from 'inquirer';
+import { DeploymentDefinition } from './types';
+import { RepositoryCollection } from '../resource/Repository/types';
+import { EnvironmentCollection } from '../resource/Environment/types';
+import { SubGenerator } from '../subgenerator';
+import { Questions } from 'yeoman-generator';
+
+const strategies: Array<ListChoiceOptions> = [
+  {
+    name: 'Capistrano',
+    value: 'capistrano',
+  },
+  {
+    name: 'Artifact Repository',
+    value: 'artifact',
+  },
+];
+
+/**
+ * A Manifest sub-generator responsible for prompting and configuration of deployments.
+ *
+ * This generator is dependent on both the Environment and Repository sub-generators as well.
+ */
+class Deployment extends SubGenerator<DeploymentDefinition> {
+  private repositories: RepositoryCollection = {};
+  private environments: EnvironmentCollection = {};
+  protected configName = 'deployments';
+  protected editPrompt = 'Would you like to update your deployments?';
+
+  _setResources(resources: Record<string, ResourceCollection>): void {
```

Comment:
Now that we've addressed the types of the subgenerators, we can fix this method. This method is only used by this generator, so we can remove the signature from the parent class (and its implementations in `Repository` and `Environment`), and either split this method into two (`_setRepositories` and `_setEnvironments`) or update the signature to be `_setResources(repositories, environments)`.